### PR TITLE
Add getopts and usage

### PR DESCRIPTION
**Pull request by jharrington22**
@jewzaam @mwoodson my attempt to make script a little safer. With these changes you can run the script without arguments and it will print the usage (still) but adding a specific option for status and upgrade versions. Thoughts?

### CODE REVIEW

**Comment by jharrington22** [comment]
I've tested the status option but not the upgrade.

File: scripts/cluster-upgrade.sh [hunk 1]
```diff
@@ -1,19 +1,75 @@
 #!/bin/bash
-# This script contains non-portable components and is intended to run on a Hive cluster.
 
-OCP_VERSION_FROM=$1
-OCP_VERSION_TO=$2
+STATUS=false
```
Comment (lisa) [inline scripts/cluster-upgrade.sh]:
```suggestion
STATUS=false
set -e
```

little safer

Comment (jharrington22) [reply scripts/cluster-upgrade.sh]: Good point :)

Comment (jharrington22) [reply scripts/cluster-upgrade.sh]: @lisa fixed

File: scripts/cluster-upgrade.sh [hunk 2]
```diff
@@ -1,19 +1,75 @@
-#!/bin/bash
-# This script contains non-portable components and is intended to run on a Hive cluster.
+#!/bin/bash -e
```
Comment (cblecker) [inline scripts/cluster-upgrade.sh]:
```suggestion
#!/usr/bin/env bash

set -o errexit
set -o nounset
set -o pipefail
```

File: scripts/cluster-upgrade.sh [hunk 3]
```diff
@@ -1,19 +1,75 @@
...
 
-OCP_VERSION_FROM=$1
-OCP_VERSION_TO=$2
+STATUS=false
+SCRIPT_NAME=$(basename "$0")
+
+usage() {
+    cat <<EOF
```
Comment (cblecker) [inline scripts/cluster-upgrade.sh]: Reason for cat > echo?

Comment (jharrington22) [reply scripts/cluster-upgrade.sh]: Its cleaner, otherwise I'd have to echo every line or string together `\n`'s. Do you think?

**Comment by jharrington22** [comment]
I'll squash before merge, so please don't lgtm @lisa @cblecker

File: scripts/cluster-upgrade.sh (hunk 3 above)
Comment (cblecker) [reply scripts/cluster-upgrade.sh]: Oh yeah, you're using a here doc. Yeah, that's fine then 👍

File: scripts/cluster-upgrade.sh [hunk 4]
```diff
@@ -1,19 +1,78 @@
...
+        t)
+            OSD_UPGRADE_TO="$OPTARG" >&2
+            ;;
+        s)
+            STATUS=true >&2
+            ;;
+        h)
+            echo "Invalid option: -$OPTARG" >&2
+            usage
+            exit 1
```
Comment (lisa) [inline scripts/cluster-upgrade.sh]: I don't think showing help should be an error?

Comment (jharrington22) [reply scripts/cluster-upgrade.sh]: Good pickup!

File: scripts/cluster-upgrade.sh [hunk 5]
```diff
@@ -1,19 +1,76 @@
...
+        :)
+            echo "$0 Requires an argument" >&2
+            usage
+            exit 1
+            ;;
+        esac
+    done
+
+
+OCP_VERSION_FROM=OSD_UPGRADE_FROM
```
Comment (cblecker) [inline scripts/cluster-upgrade.sh]: Why do we set this as one name from getops, and then set it to the exact same thing down here? Seems redundant.

**Comment by jewzaam** [comment]
@jharrington22 still something you'd like to merge?

**Pull request closed without merging**